src/terms/blank_node.rs

```diff
@@ -0,0 +1,46 @@
+use crate::error::Error;
+use std::fmt::{Display, Formatter};
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct BlankNode {
+    value: String,
+}
+
+impl BlankNode {
+    pub fn new(value: &str) -> Result<Self, Error> {
+        Ok(BlankNode {
+            value: value.to_string(),
+        })
+    }
+}
+
+impl Display for BlankNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "_:{}", self.value)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn equality() {
+        let bn1 = BlankNode {
+            value: "foo".to_string(),
+        };
+        let bn2 = BlankNode {
+            value: "foo".to_string(),
+        };
+
+        assert_eq!(bn1, bn2);
+        assert_eq!(bn2, bn1);
```

Comment:
You've marked `BlankNode` with `#[derive(PartialEq, *Eq*)]`, so there is no need to check for equality both ways.

Reply:
yeah - as I was learning, the test was essentially to verify that the attribute was behaving as I expected